pkg/runner/ftp.go

```diff
@@ -401,15 +369,85 @@ func (fc *FtpClient) cpDir(src, dst string) (CommandResult, error) {
 }
 
 func (fc *FtpClient) cpFile(src, dst string) (CommandResult, error) {
-	cmd := exec.Command("cp", src, dst)
-	cmd.SysProcAttr = fc.sysProcAttr
-	if output, err := cmd.CombinedOutput(); err != nil {
+	err := copyFile(src, dst)
+	if err != nil {
 		return CommandResult{
-			Message: strings.ToLower(string(output)),
+			Message: err.Error(),
 		}, err
 	}
 
 	return CommandResult{
 		Message: fmt.Sprintf("Copy %s to %s", src, dst),
 	}, nil
 }
+
+func copyFile(src, dst string) error {
+	srcFile, err := os.Open(src)
+	if err != nil {
+		return err
+	}
+	defer srcFile.Close()
+
+	dstFile, err := os.Create(dst)
+	if err != nil {
+		return err
+	}
+	defer dstFile.Close()
+
+	if _, err = io.Copy(dstFile, srcFile); err != nil {
+		return err
+	}
+
+	if err = dstFile.Close(); err != nil {
+		return err
+	}
+
+	srcInfo, err := os.Stat(src)
+	if err != nil {
+		return err
+	}
+
+	if err = os.Chmod(dst, srcInfo.Mode()); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func copyDir(src, dst string) error {
+	if strings.HasPrefix(dst, src) {
```

Comment:
Should the two paths (src and dst) be aligned to the same format? They could differ as relative vs absolute paths – is this accounted for?

Reply:
To support relative paths, `parsePath()` parses relative paths into absolute paths based on the home directory. 
`copyDir()` and other `Web FTP Commands` utilize these parsed paths.